### PR TITLE
Minor cleanup of admin pages

### PR DESCRIPTION
**Pull request by eseliger**
While going through the product to familiarize myself more with the current first time admin experience, I noticed a few smaller things that I fixed along the way and batched in this PR.

## Test plan

Tested changes look good locally.

### CODE REVIEW

File: client/web/src/components/externalServices/AddExternalServicesPage.module.scss [hunk 1]
```diff
@@ -1,33 +1,4 @@
...
-    --btn-light-bg: #f8f9fa;
-    --btn-light-hover-bg: #e2e6ea;
-    --btn-light-active-border-color: #dae0e5;
-}
 .add-external-services-page__card {
     display: block;
     margin-bottom: 1rem;
 }
-
-:global(.theme-light) .btn-light,
```
Comment (eseliger) [inline client/web/src/components/externalServices/AddExternalServicesPage.module.scss:12]: This felt absolutely excessive for just making the button look like none of our other buttons.

File: client/web/src/components/externalServices/AddExternalServicesPage.tsx [hunk 2]
```diff
@@ -49,8 +50,8 @@ export const AddExternalServicesPage: FC<AddExternalServicesPageProps> = ({
     isSourcegraphApp,
 }) => {
     const { search } = useLocation()
-    const [hasDismissedPrivacyWarning, setHasDismissedPrivacyWarning] = useLocalStorage(
-        'hasDismissedCodeHostPrivacyWarning',
+    const [hasDismissedPrivacyWarning, setHasDismissedPrivacyWarning] = useTemporarySetting(
```
Comment (eseliger) [inline client/web/src/components/externalServices/AddExternalServicesPage.tsx:53]: Make sure this persists across browser boundaries.

File: client/web/src/components/externalServices/AddExternalServicesPage.tsx [hunk 3]
```diff
@@ -99,8 +100,13 @@ export const AddExternalServicesPage: FC<AddExternalServicesPageProps> = ({
 
     return (
         <>
-            <PageTitle title="Add code host connection" />
-            <H2>Add code host connection</H2>
+            <PageTitle title="Add a code host connection" />
+            <PageHeader
```
Comment (eseliger) [inline client/web/src/components/externalServices/AddExternalServicesPage.tsx:104]: Aligns header style.

File: client/web/src/components/externalServices/AddExternalServicesPage.tsx [hunk 4]
```diff
@@ -112,9 +118,9 @@ export const AddExternalServicesPage: FC<AddExternalServicesPageProps> = ({
                     <Link to="/user/app-settings">Settings → Repositories → Local/Remote repositories</Link>
                 </LimitedAccessBanner>
             )}
-            <Container>
-                <Text>Add code host connection to one of the supported code hosts.</Text>
-                {hasDismissedPrivacyWarning && (
+
+            <Container className="mb-3">
+                {!hasDismissedPrivacyWarning && (
```
Comment (eseliger) [inline client/web/src/components/externalServices/AddExternalServicesPage.tsx:123]: bug? The notice never showed.

File: client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx [hunk 5]
```diff
@@ -297,20 +299,20 @@ export const CreateGitHubAppPage: FC<CreateGitHubAppPageProps> = ({
...
-                        Cancel
-                    </ButtonLink>
-                </div>
                 {/* eslint-disable-next-line react/forbid-elements */}
                 <form ref={ref} method="post">
                     {/* eslint-disable-next-line react/forbid-elements */}
                     <input ref={formInput} name="manifest" onChange={noop} hidden={true} />
                 </form>
             </Container>
+            <div>
```
Comment (eseliger) [inline client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx:308]: page level actions should live outside of a container

File: client/web/src/repo/settings/RepoSettingsMirrorPage.tsx [hunk 6]
```diff
@@ -294,8 +294,9 @@ const LastSyncOutputContainer: FC<LastSyncOutputProps> = props => {
         <BaseActionContainer
             title="Last sync log"
             titleAs="h3"
+            className="mb-0"
             description={<H4>Output from this repository's most recent sync</H4>}
-            details={<LogOutput text={output} logDescription="Job output:" />}
+            details={<LogOutput className="w-100" text={output} logDescription="Job output:" />}
```
Comment (eseliger) [inline client/web/src/repo/settings/RepoSettingsMirrorPage.tsx:299]: fixes unstable width with shorter log outputs

File: client/web/src/components/gitHubApps/GitHubAppsPage.tsx [hunk 7]
```diff
@@ -85,32 +85,35 @@ export const GitHubAppsPage: React.FC<Props> = ({ batchChangesEnabled }) => {
...
-                            nodes: gitHubApps ?? [],
-                            totalCount: gitHubApps?.length ?? 0,
-                        }}
-                        noun="GitHub App"
-                        pluralNoun="GitHub Apps"
-                        hasNextPage={false}
-                    />
-                </SummaryContainer>
-            </ConnectionContainer>
+            <Container className="mb-3">
```
Comment (eseliger) [inline client/web/src/components/gitHubApps/GitHubAppsPage.tsx:88]: pro tip: hide whitespaces makes this much easier to review.

File: client/web/src/components/gitHubApps/GitHubAppsPage.tsx [hunk 8]
```diff
@@ -85,32 +85,35 @@ export const GitHubAppsPage: React.FC<Props> = ({ batchChangesEnabled }) => {
...
+                    {loading && !data && <ConnectionLoading />}
+                    <ConnectionList as="ul" className="list-group" aria-label="GitHub Apps">
+                        {gitHubApps?.map(app => (
+                            <GitHubAppCard key={app.id} app={app} refetch={reloadApps} />
+                        ))}
+                    </ConnectionList>
+                    <SummaryContainer className="mt-2" centered={true}>
+                        <ConnectionSummary
+                            emptyElement={
+                                <div className="text-center text-muted">You haven't created any GitHub Apps yet.</div>
```
Comment (eseliger) [inline client/web/src/components/gitHubApps/GitHubAppsPage.tsx:101]: fixes double "no elements" message (one standard from the summary, one from the old logic to render this notice

File: client/web/src/components/gitHubApps/GitHubAppsPage.tsx (hunk 7 above)
Comment (eseliger) [reply client/web/src/components/gitHubApps/GitHubAppsPage.tsx:88]: adds container for visual consistency

File: client/web/src/repo/settings/RepoSettingsMirrorPage.tsx [hunk 9]
```diff
@@ -294,8 +294,9 @@ const LastSyncOutputContainer: FC<LastSyncOutputProps> = props => {
         <BaseActionContainer
             title="Last sync log"
             titleAs="h3"
+            className="mb-0"
```
Comment (sashaostrikov) [inline client/web/src/repo/settings/RepoSettingsMirrorPage.tsx:297]: Q: why this is needed?

**Review by sashaostrikov — APPROVED** [review]
LGTM! Asked 2 questions just to understand things better :)

File: client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx [hunk 10]
```diff
@@ -199,19 +199,21 @@ export const CreateGitHubAppPage: FC<CreateGitHubAppPageProps> = ({
...
             <PageHeader
                 path={[{ text: pageTitle }]}
+                headingElement="h2"
                 description={
                     headerDescription || (
                         <>
-                            Register a GitHub App to better manage GitHub code host connections.
-                            <Link to="/help/admin/external_service/github#using-a-github-app" className="ml-1">
+                            Register a GitHub App to better manage GitHub code host connections.{' '}
+                            <Link to="/help/admin/external_service/github#using-a-github-app" target="_blank">
```
[the remark this answers is not in the document]
Comment (kopancek) [reply client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx:207]: Open in a new tab/window.

File: client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx [hunk 11]
```diff
@@ -199,19 +199,21 @@ export const CreateGitHubAppPage: FC<CreateGitHubAppPageProps> = ({
...
                                 See how GitHub App configuration works.
                             </Link>
                         </>
                     )
                 }
                 annotation={headerAnnotation}
+                className="mb-3"
```
Comment (kopancek) [inline client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx:214]: Just curious, why we switched from mt-3 on the container to mb-3 here. Is it better to use bottom margins for some reason?

File: client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx (hunk 5 above)
Comment (kopancek) [reply client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx:308]: Is this defined in some list of conventions?

File: client/web/src/repo/settings/RepoSettingsMirrorPage.tsx (hunk 9 above)
Comment (kopancek) [reply client/web/src/repo/settings/RepoSettingsMirrorPage.tsx:297]: IIRC containers have some margins defined by default.

**Review by kopancek — COMMENTED** [review]
Can you post some screenshots for the changes, especially the github app sections? We implemented it based on designs from @meglynst , so it would probably be good if she reviewed as well.

File: client/web/src/repo/settings/RepoSettingsMirrorPage.tsx (hunk 9 above)
Comment (sashaostrikov) [reply client/web/src/repo/settings/RepoSettingsMirrorPage.tsx:297]: makes sense, haven't thought about it!

File: client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx (hunk 11 above)
Comment (eseliger) [reply client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx:214]: That's mostly just for consistency, most of our admin headers have this class applied on the PageHeader component, and it looked cleaner when looking through usages when all of them had the same class applied.

File: client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx (hunk 5 above)
Comment (eseliger) [reply client/web/src/components/gitHubApps/CreateGitHubAppPage.tsx:308]: There's some documentation in this storybook: https://storybook.sgdev.org/?path=/story/wildcard-container--overview

**Comment by eseliger** [comment]
> Can you post some screenshots for the changes, especially the github app sections? We implemented it based on designs from @meglynst , so it would probably be good if she reviewed as well.

Yep!

Previous:
Screenshot 2023-07-20 at 02 17 18@2x

After:
Screenshot 2023-07-20 at 02 19 21@2x

Previous:
Screenshot 2023-07-20 at 02 21 23@2x

After:
Screenshot 2023-07-20 at 02 19 56@2x

**Pull request merged**